ui/packages/catalog-ui-search/src/main/webapp/component/result-item/result-item.view.js

```diff
@@ -443,19 +455,27 @@ const ResultItemView = Marionette.LayoutView.extend({
         .get('associations.external') !== undefined
     )
   },
-  checkDisplayType: function() {
-    var displayType = user
-      .get('user')
-      .get('preferences')
-      .get('resultDisplay')
+  checkDisplayType() {
+    const displayType = getResultDisplayType()
+
     switch (displayType) {
-      case 'List':
+      case LIST_DISPLAY_TYPE:
         this.$el.removeClass('is-grid').addClass('is-list')
         break
-      case 'Grid':
+      case GRID_DISPLAY_TYPE:
         this.$el.addClass('is-grid').removeClass('is-list')
         break
     }
+
+    //Only re-render when going from `List` to `Grid`
+    if (
+      this.currentDisplayType !== displayType &&
```

Comment:
✏️ 	Maybe rename `displayType` to `incomingDIsplayType` or something similar to make it more clear.